### DISCUSSION TITLE
tutorials/intro-typescript/10-query-arguments

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#102] [course feedback]
# tutorials/intro-typescript/10-query-arguments

https://www.apollographql.com/tutorials/intro-typescript/10-query-arguments

**Comment by moishinetzer** [comment]
`concurrently` needs to be added to the `package.json`

**Reply by addysia** [reply]
for noobs, that means adding the package "npm i -D concurrently" too.

**Reply by mabuyo** [reply]
Good catch! We'll update the starter repo shortly, thank you!

**Comment by justinkelly69** [comment]
When I run the query I get a CSRF error. How to fix this?

**Reply by mabuyo** [reply]
Hey there @justinkelly69! Could you clarify where you're seeing the error? Is it on Sandbox, or your server terminal, or browser console?

**Comment by justinkelly69** [comment]
It's in the sandbox when I run localhost:4000

{
"errors": [
{
"message": "This operation has been blocked as a potential Cross-Site
Request Forgery (CSRF). Please either specify a 'content-type' header (with
a type that is not one of application/x-www-form-urlencoded,
multipart/form-data, text/plain) or provide a non-empty value for one of
the following headers: x-apollo-operation-name, apollo-require-preflight\n",
"extensions": {
"code": "BAD_REQUEST",
"stacktrace": [
"BadRequestError: This operation has been blocked as a potential Cross-Site
Request Forgery (CSRF). Please either specify a 'content-type' header (with
a type that is not one of application/x-www-form-urlencoded,
multipart/form-data, text/plain) or provide a non-empty value for one of
the following headers: x-apollo-operation-name, apollo-require-preflight",
"",
" at new GraphQLErrorWithCode
***@***.***/server/src/internalErrorClasses.ts:15:5)"
,
" at new BadRequestError
***@***.***/server/src/internalErrorClasses.ts:116:5)"
,
" at preventCsrf
***@***.***/server/src/preventCsrf.ts:91:9)"
,
" at ApolloServer.executeHTTPGraphQLRequest
***@***.***/server/src/ApolloServer.ts:1052:20)"
]
}
}
]
}

**Reply by mabuyo** [reply]
Hmm, what browser are you using?

**Comment by justinkelly69** [comment]
chrome on OSX M1 and Linux amd64

**Comment by sunnytoolsmiths** [comment]
I run npm dev run, in the terminal, it says the server is running but in the browser it gives the error. Can you please help me understand what issue it could be?

**Reply by sunnytoolsmiths** [reply]
actually I realized that I need to add / in the end of the localhost url... (issue resolved)

**Comment by mark-P0** [comment]
One of the `--watch` arguments in the new `dev` and `generate` scripts can be omitted, yes?
